TestOpt4bitBnb::test_lora_4bit result mismatch

### System Info

latest main branch

### Who can help?

@BenjaminBossan 

### Reproduction

`CUDA_VISIBLE_DEVICES=5,6 pytest --regression tests/regression/test_regression.py::TestOpt4bitBnb::test_lora_4bit`
failed on A100

output:
```
>           assert torch.allclose(output_loaded, output, atol=self.tol, rtol=self.tol)                                              E           AssertionError: assert False
E            +  where False = <built-in method allclose of type object at 0x7f197fcb7c40>(tensor([[[-3.1559,  0.3479,  2.9764,  ...,
 -2.6444, -4.4152,  1.1950],\n         [ 1.4434,  1.7568,  4.9091,  ..., -1.9...11, -2.5819,  2.8310],\n         [ 0.7241,  2.6541,
 5.2471,  ..., -1.4358, -2.7339,  2.4805]]],\n       device='cuda:0'), tensor([[[-3.1605,  0.3543,  2.9821,  ..., -2.6416, -4.4117,
 1.2042],\n         [ 1.4392,  1.7604,  4.9075,  ..., -1.9...95, -2.5805,  2.8382],\n         [ 0.7251,  2.6641,  5.2492,  ..., -1.4
326, -2.7258,  2.4895]]],\n       device='cuda:0'), atol=0.0001, rtol=0.0001)
E            +    where <built-in method allclose of type object at 0x7f197fcb7c40> = torch.allclose
E            +    and   0.0001 = <tests.regression.test_regression.TestOpt4bitBnb testMethod=test_lora_4bit>.tol
E            +    and   0.0001 = <tests.regression.test_regression.TestOpt4bitBnb testMethod=test_lora_4bit>.tol

tests/regression/test_regression.py:260: AssertionError
```

### Expected behavior

I'd like to update the ground truth, would like to hear your opinion. Thanks!

--- a/tests/regression/test_regression.py
+++ b/tests/regression/test_regression.py
@@ -669,7 +669,7 @@ class TestOpt4bitBnb(RegressionTester):
         model = get_peft_model(base_model, config)
         # NVIDIA A100 requires a lower tol to pass the test.
         old_tol = self.tol
-        self.tol = 5e-3
+        self.tol = 3e-2
         try:
             self.assert_results_equal_or_store(model, LORA_4BIT_FOLDER)
         finally:

